main/src/ca/uwaterloo/flix/language/errors/RedundancyError.scala

```diff
@@ -312,15 +312,17 @@ object RedundancyError {
   /**
     * An error raised to indicate that an expression is useless.
     *
+    * @param tpe the type of the expression.
     * @param loc the location of the expression.
     */
-  case class UselessExpression(loc: SourceLocation) extends RedundancyError {
+  case class UselessExpression(tpe: Type, loc: SourceLocation) extends RedundancyError {
     def summary: String = "Useless expression."
 
     def message(formatter: Formatter): String = {
       import formatter._
       s"""${line(kind, source.name)}
          |>> Useless expression: It has no side-effect(s) and its result is discarded.
+         |>> The expression has type '${FormatType.formatWellKindedType(tpe)}'
```

Comment:
This is some extra information. I would move this after the code part (and remove the >>).